patroni/utils.py

```diff
@@ -416,3 +417,22 @@ def cluster_as_json(cluster):
         if cluster.failover.candidate:
             ret['scheduled_switchover']['to'] = cluster.failover.candidate
     return ret
+
+
+def directory_exists_or_create(d, msg="{} {}"):
+    try:
+        if not os.path.exists(d):
+            os.makedirs(d)
+        if os.path.isdir(d):
+            test_file = os.path.join(d, ".writable")
+            try:
+                test_writable = open(test_file, "w")
```

Comment:
I would better use here:
```python
fd, tmpfile = tempfile.mkstemp(dir=d)
os.close(fd)
os.remove(tmpfile)
```

In case of problems it raises `OSError`.